yt_dlp/extractor/radlive.py

```diff
@@ -0,0 +1,422 @@
+import json
+import re
+import textwrap
+import urllib.request
+from datetime import datetime
+
+from ..utils import ExtractorError
+from .common import InfoExtractor
+
+
+class RadLiveIE(InfoExtractor):
+    IE_NAME = 'radlive'
+    _VALID_URL = r'https?://(?:www\.)?rad\.live/content/(?P<content_type>feature|episode)/(?P<id>[a-f0-9-]+)'
+    _TESTS = [{
+        'url': 'https://rad.live/content/feature/dc5acfbc-761b-4bec-9564-df999905116a',
+        'md5': '6219d5d31d52de87d21c9cf5b7cb27ff',
+        'info_dict': {
+            'id': 'dc5acfbc-761b-4bec-9564-df999905116a',
+            'ext': 'mp4',
+            'title': 'Deathpact - Digital Mirage 2 [Full Set]',
+            'language': 'en',
+            'thumbnail': 'https://static.12core.net/cb65ae077a079c68380e38f387fbc438.png',
+            'description': '',
+            'release_timestamp': 1600185600.0,
+            'channel': 'Proximity',
+            'channel_id': '9ce6dd01-70a4-4d59-afb6-d01f807cd009',
+            'channel_url': 'https://rad.live/content/channel/9ce6dd01-70a4-4d59-afb6-d01f807cd009',
+        }
+    }, {
+        'url': 'https://rad.live/content/episode/bbcf66ec-0d02-4ca0-8dc0-4213eb2429bf',
+        'md5': '40b2175f347592125d93e9a344080125',
+        'info_dict': {
+            'id': 'bbcf66ec-0d02-4ca0-8dc0-4213eb2429bf',
+            'ext': 'mp4',
+            'title': 'E01: Bad Jokes 1',
+            'language': 'en',
+            'thumbnail': 'https://lsp.littlstar.com/channels/WHISTLE/BAD_JOKES/SEASON_1/BAD_JOKES_101/poster.jpg',
+            'description': 'Bad Jokes - Champions, Adam Pally, Super Troopers, Team Edge and 2Hype',
+            'release_timestamp': None,
+            'channel': None,
+            'channel_id': '',
+            'channel_url': None,
+        },
+    }]
+
+    def _real_extract(self, url):
+        m = re.fullmatch(self._VALID_URL, url)
+
+        video_id = m.group('id')
+        content_type = m.group('content_type')
+
+        webpage = self._download_webpage(url, video_id)
+
+        _info = json.loads(self._search_regex(
+            r'<script[^>]*type=([\'"])application/json\1[^>]*>(?P<json>{.+?})</script>',
+            webpage, 'video info', group='json'))['props']['pageProps']['initialContentData']
+        info = _info[content_type]
+
+        if not info:
+            raise ExtractorError('Unable to extract video info')
+
+        formats = self._extract_m3u8_formats(info['assets']['videos'][0]['url'], video_id)
+        self._sort_formats(formats)
+
+        data = info.get('structured_data', {})
+
+        release_date = data.get('releasedEvent', {}).get('startDate')
+        if release_date:
+            try:
+                release_date = datetime.strptime(release_date, '%Y-%m-%dT%H:%M:%S.%f%z').timestamp()
+            except ValueError:
+                release_date = None
+
+        channel = next(iter(_info.get('channels', [])), {})
+        channel_id = channel.get('lrn', '').split(':')[-1]
```

Comment:
```suggestion
        channel_id = channel.get('lrn', '').split(':')[-1] or None
```
This will make sure it's `None` if `''`

Reply:
Applied manually, also updated the test.